src/audio/buffer.c

```diff
@@ -119,13 +131,19 @@ void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 	 * 4. source(non-DMA) --> buffer --> sink(non-DMA): do nothing.
 	 */
 	if (buffer->source->is_dma_connected &&
-	    !buffer->sink->is_dma_connected)
+	    !buffer->sink->is_dma_connected) {
 		/* need invalidate cache for sink component to use */
-		dcache_invalidate_region(buffer->w_ptr, bytes);
+		dcache_invalidate_region(head, head_size);
+		if (tail_size > 0)
```

Comment:
Just to make checkpatch happy:
`if (tail_size)`

Reply:
@tlauda I laughed at my desk when I saw my trademarked wording "make checkpatch happy" is now used by the firmware team :-)